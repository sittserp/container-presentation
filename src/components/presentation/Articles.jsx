import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const Articles = ({ articles }) => {
  const articleElements = articles.map(article => (
    <li key={article.id}>
      <Article {...article} />
    </li>
  ));

  return (
    <ul>
      {articleElements}
    </ul>
  );
};

Articles.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    author: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    urlToImage: PropTypes.string.isRequired
  })).isRequired
};

export default Articles;
