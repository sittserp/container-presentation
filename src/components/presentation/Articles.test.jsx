import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Articles from './Articles';

describe('Articles component', () => {
  afterEach(() => cleanup());
  it('renders Articles', () => {
    const { asFragment } = render(<Articles
      articles={[]} 
      author="dummy data author" 
      title="dummy data title"
      description="dummy data description"
      url="dummy data url"
      urlToImage="dummy data urlToImage"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
