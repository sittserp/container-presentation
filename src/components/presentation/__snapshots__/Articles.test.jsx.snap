// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles component renders Articles 1`] = `
<DocumentFragment>
  <ul
    data-testid="display"
  />
</DocumentFragment>
`;
